ttnn/cpp/ttnn/operations/moreh/moreh_adam/device/moreh_adam_device_operation.hpp

```diff
@@ -0,0 +1,99 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <variant>
+#include <vector>
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/device_operation.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+#include "ttnn/tensor/types.hpp"
+
+namespace ttnn::operations::moreh::moreh_adam {
+struct MorehAdamOperation {
+    struct operation_attributes_t {
+        float lr;
+        float beta1;
+        float beta2;
+        float eps;
+        float weight_decay;
+        uint32_t step;
+        bool amsgrad;
+
+        const std::optional<MemoryConfig>& output_mem_config;
```

Comment:
```suggestion
        const MemoryConfig output_memory_config;
```
https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_device_operation.hpp#L35

"Looking at other code, it seems that MemoryConfig is used instead of optional<MemoryConfig>."

Reply:
No, the actual issue is that you can (and arguably, shouldn't, since it is already `optional`) define this as `const std::optional<MemoryConfig>&`

Using `const std::optional<MemoryConfig>` works just fine and is passing tests https://github.com/tenstorrent/tt-metal/actions/runs/10694588733